File: support/catdb.pl
#!/usr/bin/env perl

###
## @(#) catdb.pl    falcon@icb.net  2/1/98
## Updated by hoche@grok.com       5/24/19
##
## just cat the entire damn database to stdout
###

use strict;
use warnings;

dbmopen(my %DB, "./icbdb", 0666);

while (my ($key,$val) = each(%DB)) {
    printf("%s = %s\n", $key, $val);
}

dbmclose(%DB);

File: support/catdb.py
#!/usr/bin/env python3

###
## catdb.py    falcon@icb.net  2/1/98
##
## just cat the entire damn database to stdout
###

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
from icbdb import IcbDb

def main():
    dbpath = sys.argv[1] if len(sys.argv) > 1 else "./icbdb"
    with IcbDb(dbpath) as db:
        for key, val in db.items():
            print(f"{key} = {val}")

if __name__ == "__main__":
    main()
